src/fs/rock/RockRebuild.cc

```diff
@@ -667,9 +676,14 @@ Rock::Rebuild::freeBadEntry(const sfileno fileno, const char *eDescription)
 void
 Rock::Rebuild::swanSong()
 {
-    debugs(47,3, HERE << "cache_dir #" << sd->index << " rebuild level: " <<
-           StoreController::store_dirs_rebuilding);
-    storeRebuildComplete(&counts);
+    debugs(47, 3, "cache_dir #" << sd->index);
+    storeRebuildComplete(&counts, *sd);
+    if (opt_foreground_rebuild)
+        sd->startAcceptingRequests();
+    if (UsingSmp()) {
+        assert(IamDiskProcess());
+        Ipc::StrandMessage::NotifyCoordinator(Ipc::mtRebuildFinished, sd->filePath);
+    }
```

Comment:
No, it can't, because IpcIoFile (and its base) has nothing about Disk. It passes its IpcIoFile::dbName to Coordinator (which is copied from filePath in its constructor). And we would not be able to pass Disk::tag() to IpcIoFile (instead of filePath) because this is the disk file it works with. We could theoretically add a second 'tag' parameter to IpcIoFile constructor (and the whole hierarchy), but I am not sure this is the right approach.